plugins/fabric8-cdn-support/src/main/java/com/redhat/che/cdnSupport/CdnSupportService.java

```diff
@@ -0,0 +1,315 @@
+  private static final ObjectMapper YAML_PARSER = new ObjectMapper(new YAMLFactory());
+  private static final ObjectMapper JSON_PARSER = new ObjectMapper(new JsonFactory());
+  private static final String SKOPEO_BINARY = "skopeo";
+  private static final String[] SKOPEO_HELP_ARGS = new String[] {"--help"};
+  private static final long SKOPEO_HELP_TIMEOUT_SECONDS = 2;
+  private static final String SKOPEO_INSPECT_ARG = "inspect";
+  private static final String SKOPEO_IMAGE_PREFIX = "docker://";
+  private static final long SKOPEO_INSPECT_TIMEOUT_SECONDS = 5;
+  private static final String TAR_BINARY = "tar";
+  private static final String TAR_ARG = "-xvf";
+  private static final long TAR_TIMEOUT_SECONDS = 5;
+  private static final String CHE_PLUGIN_YAML_FILE_NAME = "che-plugin.yaml";
+  private static final String CHE_PLUGIN_TGZ_FILE_NAME = "che-plugin.tar.gz";
+  private static final String TEMP_DIR_PREFIX = "che-plugin-archive-dir";
+  private static final String LABEL_NAME = "che-theia-plugin.cdn.artifacts";
+
+  private static final CompletableFuture<?>[] FUTURE_ARRAY = new CompletableFuture<?>[0];
+  private static final CommandRunner RUNNER = new CommandRunner();
+
+  private final String editorToPrefetch;
+  private final CommandRunner commandRunner;
+  private final PluginMetaRetriever pluginMetaRetriever;
+  @VisibleForTesting String editorDefinitionUrl = null;
+  @VisibleForTesting String dockerImage = null;
+
+  @Inject
+  public CdnSupportService(
+      PluginMetaRetriever pluginMetaRetriever,
+      @Nullable @Named("che.fabric8.cdn.prefetch.editor") String editorToPrefetch) {
+    this(RUNNER, pluginMetaRetriever, editorToPrefetch);
+  }
+
+  CdnSupportService(
+      CommandRunner commandRunner,
+      PluginMetaRetriever pluginMetaRetriever,
+      @Nullable @Named("che.fabric8.cdn.prefetch.editor") String editorToPrefetch) {
+    this.pluginMetaRetriever = pluginMetaRetriever;
+    this.editorToPrefetch = editorToPrefetch;
+    this.commandRunner = commandRunner;
+
+    // Test that the skopeo process is available
+
+    try {
+      commandRunner.runCommand(
+          SKOPEO_BINARY,
+          SKOPEO_HELP_ARGS,
+          null,
+          SKOPEO_HELP_TIMEOUT_SECONDS,
+          TimeUnit.SECONDS,
+          null,
+          null);
+    } catch (IOException e) {
+      throw new RuntimeException(
+          "The `skopeo` command is not available. Please chack that is has been correctly installed in the Che server Docker image",
+          e);
+    } catch (ExecutionException | InterruptedException | TimeoutException e) {
+      LOG.warn("Exception during the `skopeo --help` command execution", e);
+    }
+  }
+
+  @GET
+  @Path("paths")
+  @Produces(APPLICATION_JSON)
+  public String getPaths(@Context HttpServletRequest servletRequest) throws Exception {
+    final String empty = "{}";
+    if (editorToPrefetch == null) {
+      return empty;
```

Comment:
should it be empty or 404 ?
AFAIK from client, it should only provide result or an error code, no ?

Reply:
I changed the code accordingly